spec/controllers/attachments_controller_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+RSpec.describe AttachmentsController, type: :controller do
+  describe 'Filters' do
+    it { should use_before_action(:authenticate_user!) }
+  end
+
+  describe 'GET #index' do
+    context 'when the user has at least one attachment' do
+      it 'redirects to the last attachment detail' do
+        user = create :user
+        create_list :attachment, 10, user_id: user.id
+        last_attachment = user.attachments.last
+
+        sign_in user
+        get :index
+
+        expect(response).to redirect_to results_attachment_path(last_attachment)
+      end
+    end
+
+    context 'when the user does not have any attachments' do
+      it 'renders the page with an instantiated instance variable' do
+        user = create :user
+
+        sign_in user
+        get :index
+
+        expect(subject).to render_template(:index)
+        expect(assigns(:attachment)).to be_a(Attachment)
+        expect(assigns(:attachment)).to be_new_record
+      end
+    end
+  end
+
+  describe 'GET #results' do
+    it 'renders the page with instance variables' do
+      user = create :user
+      attachments = create_list :attachment, 10, user_id: user.id
+      attachment = attachments.sample
+
+      sign_in user
+      get :results, params: { id: attachment.id }
+
+      expect(response).to render_template(:results)
+      expect(assigns(:attachments)).to eq user.attachments.order(created_at: :desc)
+      expect(assigns(:attachment)).to eq attachment
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when upload valid file' do
```

Comment:
could we rewrite it for better readability?
```suggestion
    context 'when a valid file is uploaded' do
```